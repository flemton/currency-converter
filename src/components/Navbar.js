import { NavLink } from 'react-router-dom';

const Navbar = () => (
  <nav className="nav">
    <NavLink to="/">
      <h1 className="homepage">{'<'}</h1>
    </NavLink>
    <NavLink to="/">
      <h1 className="header">Currency Converter</h1>
    </NavLink>
  </nav>
);

export default Navbar;
