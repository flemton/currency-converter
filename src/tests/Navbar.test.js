import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';
import '@testing-library/jest-dom/extend-expect';

describe('Test suite for the Navbar component', () => {
  let navbar;
  beforeEach(() => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    navbar = screen.getByText(/Currency Converter/i);
  });

  it('Should test if the Navbar component is inside the document', () => {
    expect(navbar).toBeInTheDocument();
  });

  it('Should render a snapshot of the Navbar component', () => {
    expect(navbar).toMatchSnapshot();
  });
});
