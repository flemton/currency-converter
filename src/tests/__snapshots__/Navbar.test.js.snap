// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for the Header component Should render a snapshot of the Header component 1`] = `
<h1
  class="header"
>
  Currency Converter
</h1>
`;

exports[`Test suite for the Navbar component Should render a snapshot of the Header component 1`] = `
<h1
  class="header"
>
  Currency Converter
</h1>
`;

exports[`Test suite for the Navbar component Should render a snapshot of the Navbar component 1`] = `
<h1
  class="header"
>
  Currency Converter
</h1>
`;
